src/store/src/Document/TextDocument.php

```diff
@@ -33,4 +33,14 @@ public function withContent(string $content): self
     {
         return new self($this->id, $content, $this->metadata);
     }
+
+    public function getId(): Uuid|string|int
+    {
+        return $this->id;
+    }
+
+    public function getContent(): string
```

Comment:
lets also add a getMetadata() here, not to the interface, and make the properties private, this is anyway better for BC handling in the future.